integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java

```diff
@@ -47,42 +46,51 @@ class AppRoleRestApi extends VaultRestApi {
     static Builder appRoleBuilder() {
         return new Builder();
     }
+
     @Override
-    protected Single<WebClientRequestBuilder> updateRequestBuilderCommon(WebClientRequestBuilder requestBuilder,
-                                                                         String path,
-                                                                         ApiRequest<?> request,
-                                                                         Http.Method method,
-                                                                         String requestId) {
+    protected Http1ClientRequest updateRequestBuilderCommon(Http1ClientRequest requestBuilder,
+                                                            String path,
+                                                            ApiRequest<?> request,
+                                                            Http.Method method,
+                                                            String requestId) {
         VaultTokenBase currentToken = this.currentToken.get();
 
         if (currentToken != null) {
             if (!currentToken.renewable() || currentToken.created().plus(currentToken.leaseDuration()).isAfter(Instant.now())) {
-                requestBuilder.headers().add("X-Vault-Token", currentToken.token());
-                return Single.just(requestBuilder);
+                requestBuilder.headers(headers -> {
+                    headers.set(Http.Header.create("X-Vault-Token"), currentToken.token());
+                    return headers;
+                });
+                return requestBuilder;
             }
         }
 
-        // we need to renew the token - this may be a concurrent operation as we do not care who wins
-        return auth.login(Login.Request.create(appRoleId, secretId))
-                .map(it -> {
-                    VaultTokenBase token = it.token();
-                    this.currentToken.set(token);
-                    requestBuilder.headers().add("X-Vault-Token", token.token());
-                    return requestBuilder;
-                })
-                .onError(throwable -> LOGGER.log(Level.WARNING, "Failed to renew AppRole token", throwable));
+        try {
+            // we need to renew the token - this may be a concurrent operation as we do not care who wins
+            Login.Response response = auth.login(Login.Request.create(appRoleId, secretId));
+            VaultTokenBase token = response.token();
+            this.currentToken.set(token);
+            requestBuilder.headers(headers -> {
```

Comment:
We should make sure that we add a shortcut for this scenario.
Adding a single header with string/string should not be that complicated.

Reply:
Actually this is exactly the case where we SHOULD NEVER use `String` here. Please create a constant `HeaderName` for `X-Vault-Token` and use it everywhere instead of using a shortcut

Reply:
And of course if we are missing a `header(HeaderName, String)` and `header(HeaderValue)` on request builder, it would be good to add them

Reply:
I had added new method `header(HeaderName, String)` ; and using it here now.